androidHyperskillApp/src/main/java/org/hyperskill/app/android/stage_implementation/view/fragment/StageStepWrapperFragment.kt

```diff
@@ -102,12 +105,13 @@ class StageStepWrapperFragment :
             }
         }
         viewBinding.stageImplementationTitle.text = stageTitle
-        StepDelegate.init(viewBinding.stageImplementationError, stepViewModel::onNewMessage)
+        stepDelegate?.init(viewBinding.stageImplementationError, stepViewModel::onNewMessage)
     }
 
     override fun onDestroyView() {
         super.onDestroyView()
         viewStateDelegate = null
+        stepDelegate = null
```

Comment:
As `stepDelegate` is initialized in `onCreate`, it should be destroyed in `onDestroy`. Otherwise, after `onDestroyView` it would not be created again. See more about fragment lifecycle [here] (https://developer.android.com/guide/fragments/lifecycle#states)
The same problem in the StepFragment

Reply:
Thank you, I dove into that topic, forgot about difference between fragment and fragment view lifecycle